lib/bytes/CMakeLists.txt

```diff
@@ -11,7 +11,21 @@ add_library(${CURRENT_LIB_NAME} ${LIB_HEADERS} ${LIB_SOURCES})
 add_dependencies(${CURRENT_LIB_NAME} tls_syntax)
 target_link_libraries(${CURRENT_LIB_NAME} tls_syntax)
 target_include_directories(${CURRENT_LIB_NAME}
-  PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/include
+  PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    $<INSTALL_INTERFACE:include/${PROJECT_NAME}-${PROJECT_VERSION}>
```

Comment:
It seems like this would result in things of the form `#include <mlspp-${VERSION}/header.h>`.  Is that right?  I don't love that pattern; is there a way we could do versioning at a higher level, so that it could be handled at configuration time instead of at the `#include` stage?

Reply:
The version is just window dressing.  It does not make it easier or harder if it there.  It might make it harder if you are looking for a specific version.  There is nothing else in the code that cares about version or supports versions, so I think it is confusing.  If we need to support multiple version, we can address it then.

To answer you question, the following is what the install directory looks like.

```
.
├── mlspp-0.1
│·· ├── bytes
│·· │·· └── bytes.h
│·· ├── hpke
│·· │·· ├── certificate.h
│·· │·· ├── digest.h
│·· │·· ├── hpke.h
│·· │·· ├── random.h
│·· │·· └── signature.h
│·· ├── mls
│·· │·· ├── common.h
│·· │·· ├── core_types.h
│·· │·· ├── credential.h
│·· │·· ├── crypto.h
│·· │·· ├── key_schedule.h
│·· │·· ├── messages.h
│·· │·· ├── session.h
│·· │·· ├── state.h
│·· │·· ├── tree_math.h
│·· │·· └── treekem.h
│·· ├── mls_vectors
│·· │·· └── mls_vectors.h
│·· ├── tls
│·· │·· ├── compat.h
│·· │·· └── tls_syntax.h
│·· ├── variant.hpp
│·· └── version.h
```

Reply:
You can best see this when you install via vcpkg...

```
${PROJECT_DIR}/build/vcpkg_installed/arm64-osx/include/
```